addon/utils/disposable.js

```diff
@@ -0,0 +1,86 @@
+import { assert } from '@ember/debug';
+
+export let registeredDisposables = new WeakMap();
+
+let registeredTimers = new WeakMap();
```

Comment:
I think this should ultimately move to where we define the functional `runTask` method, in the interim lets move it into `addon/mixins/run-task.js` as it will simplify logic there quite a bit...

Reply:
👍